src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -526,6 +543,29 @@ void ReadPerfSampleIdAll(PerfEventRingBuffer* ring_buffer, const perf_event_head
   std::memcpy(&sched_wakeup, res.raw_data.get(), sizeof(sched_switch_tracepoint));
 
   ring_buffer->SkipRecord(header);
+
+  // If we did not received the necessary data for a callstack, there is no need to return a
+  // SchedSwitchWithStackPerfEvent.
+  if (res.dyn_size == 0 || res.stack_data == nullptr || res.regs == nullptr) {
+    return SchedSwitchPerfEvent{
+        .timestamp = res.time,
+        .ordered_stream = PerfEventOrderedStream::FileDescriptor(ring_buffer->GetFileDescriptor()),
+        .data =
+            {
+                .cpu = res.cpu,
+                // As the tracepoint data does not include the pid of the process that the thread
+                // being switched out belongs to, we use the pid set by perf_event_open in the
+                // corresponding generic field of the PERF_RECORD_SAMPLE.
+                // Note, though, that this value is -1 when the switch out is caused by the thread
+                // exiting. This is not the case for data.prev_pid, whose value is always correct
+                // as it comes directly from the tracepoint data.
```

Comment:
Same here: consider not repeating the whole comment, but a small comment referring you to the other comment.

Reply:
Done.